ozaria/site/views/play/level/modal/CapstoneProgressModal.js

```diff
@@ -0,0 +1,43 @@
+import ModalComponent from 'app/views/core/ModalComponent'
+import CapstoneProgressComponent from './CapstoneProgressComponent.vue'
+
+class CapstoneProgressModal extends ModalComponent {
+  // Runs before the constructor is called.
+  initialize () {
+    this.propsData = {
+      levelSlug: null,
+      capstoneStage: null,
+      remainingGoals: null
+    }
+  }
+
+  constructor (options) {
+    super(options)
+    if (options) {
+      this.propsData = {
+        levelSlug: options.levelSlug,
+        capstoneStage: options.capstoneStage,
+        remainingGoals: options.remainingGoals
+      }
+    }
+  }
+
+  destroy () {
+    if (this.onDestroy) {
+      this.onDestroy()
+    }
+
+    this.goToCinematic()
```

Comment:
Is this a placeholder for navigation in the future. If so, should this be in the destroy method?

I can see that this link is also calculated within the `CapstoneProgressComponent`. Would it be advantageous to pass in the `goToCinematic` function as a prop.

Reply:
Currently navigation appears to occur like so.

User gets the `CapstoneProgressComponent.vue`. They click the "Onwards" anchor.
This either routes them to the url. Or in the destruction of the page the `goToCinematic` method is called routing them.

We should prefer one or the other unless there is a reason for both.

Reply:
There is a bad state the level can get into where the state of goals are not properly propagated everywhere. This happens if goals progress while staying on the same page. 

To enforce that the user always goes to the cinematic before coming back, it's in the `destroy()`. But yes! This could be passed as a prop 👍